src/binder_cbs.c

```diff
@@ -180,50 +191,85 @@ binder_cbs_set_config(
     ofono_cbs_set_cb_t cb,
     void* data)
 {
-    /* setGsmBroadcastConfig(int32_t serial, vec<GsmBroadcastSmsConfigInfo>); */
     GBinderWriter writer;
+    guint32 code = self->interface_aidl == RADIO_MESSAGING_INTERFACE ?
+        RADIO_MESSAGING_REQ_SET_GSM_BROADCAST_CONFIG :
+        RADIO_REQ_SET_GSM_BROADCAST_CONFIG;
     RadioRequest* req = radio_request_new2(self->g,
-        RADIO_REQ_SET_GSM_BROADCAST_CONFIG, &writer,
+        code, &writer,
         binder_cbs_set_config_cb,
         binder_cbs_callback_data_free,
         binder_cbs_callback_data_new(self, cb, data));
 
-    GBinderParent parent;
-    GBinderHidlVec* vec = gbinder_writer_new0(&writer, GBinderHidlVec);
-    RadioGsmBroadcastSmsConfig* configs = NULL;
     char** list = topics ? g_strsplit(topics, ",", 0) : NULL;
     const guint count = gutil_strv_length(list);
-    guint i;
-
-    vec->count = count;
-    vec->owns_buffer = TRUE;
-    vec->data.ptr = configs = gbinder_writer_malloc0(&writer,
-        sizeof(RadioGsmBroadcastSmsConfig) * count);
-
-    for (i = 0; i < count; i++) {
-        RadioGsmBroadcastSmsConfig* config = configs + i;
-        const char* entry = list[i];
-        const char* delim = strchr(entry, '-');
-
-        config->selected = TRUE;
-        config->toCodeScheme = 0xff;
-        if (delim) {
-            char** range = g_strsplit(entry, "-", 0);
-
-            config->fromServiceId = atoi(range[0]);
-            config->toServiceId = atoi(range[1]);
-            g_strfreev(range);
-        } else {
-            config->fromServiceId = config->toServiceId = atoi(entry);
+
+    if (self->interface_aidl == RADIO_AIDL_INTERFACE_NONE) {
+        /* setGsmBroadcastConfig(int32_t serial, vec<GsmBroadcastSmsConfigInfo>); */
+        GBinderParent parent;
+        GBinderHidlVec* vec = gbinder_writer_new0(&writer, GBinderHidlVec);
+        RadioGsmBroadcastSmsConfig* configs = NULL;
+        guint i;
+
+        vec->count = count;
+        vec->owns_buffer = TRUE;
+        vec->data.ptr = configs = gbinder_writer_malloc0(&writer,
+            sizeof(RadioGsmBroadcastSmsConfig) * count);
+
+        for (i = 0; i < count; i++) {
+            RadioGsmBroadcastSmsConfig* config = configs + i;
+            const char* entry = list[i];
+            const char* delim = strchr(entry, '-');
+
+            config->selected = TRUE;
+            config->toCodeScheme = 0xff;
+            if (delim) {
+                char** range = g_strsplit(entry, "-", 0);
```

Comment:
Same here, do these entries _always_ have two entries when there is a delimiter?